lib/Conversion/TorchToLinalg/DataMovement.cpp

```diff
@@ -244,6 +244,294 @@ class ConvertAtenReflectionPad1dOp
+
+    // Compute the offsets for extracting the slice from the
+    // input. We need to skip the row or column through which
+    // the tile should be reflected, if any (none for the center tile).
+    Value extractHOffset[3];
+    extractHOffset[LEFT] = one;
+    extractHOffset[HCENTER] = zero;
+    extractHOffset[RIGHT] = createSubs({hDimSize, tileWidth[RIGHT], one});
+
+    Value extractVOffset[3];
+    extractVOffset[TOP] = one;
+    extractVOffset[VCENTER] = zero;
+    extractVOffset[BOTTOM] = createSubs({vDimSize, tileHeight[BOTTOM], one});
+
+    // Compute the horizontal and vertical offsets for inserting
+    // the tiles in the resultTensor.
+    Value insertHOffset[3];
+    insertHOffset[LEFT] = zero;
+    insertHOffset[HCENTER] = tileWidth[LEFT];
+    insertHOffset[RIGHT] = createAdd(hDimSize, tileWidth[LEFT]);
+
+    Value insertVOffset[3];
+    insertVOffset[TOP] = zero;
+    insertVOffset[VCENTER] = tileHeight[TOP];
+    insertVOffset[BOTTOM] = createAdd(vDimSize, tileHeight[TOP]);
+
+    auto shouldHReflect = [](PadHLoc l) { return l == LEFT || l == RIGHT; };
+    auto shouldVReflect = [](PadVLoc l) { return l == TOP || l == BOTTOM; };
+
+    SmallVector<utils::IteratorType> iteratorTypes{
+        numDims, utils::IteratorType::parallel};
+    auto idMap = AffineMap::getMultiDimIdentityMap(numDims, context);
+    SmallVector<Value> allOneStrides(numDims, one);
+
+    auto createTile = [&](PadVLoc verticalPos, PadHLoc horizontalPos) {
+      // Create the tile by extracting a slice from the input tenor.
+      SmallVector<Value> extractShape{inputShape};
+      extractShape[hDim] = tileWidth[horizontalPos];
+      extractShape[vDim] = tileHeight[verticalPos];
+
+      SmallVector<Value> extractOffsets(numDims, zero);
+      extractOffsets[hDim] = extractHOffset[horizontalPos];
+      extractOffsets[vDim] = extractVOffset[verticalPos];
+
+      Value tile = rewriter.create<tensor::ExtractSliceOp>(
+          loc, input, extractOffsets, extractShape, allOneStrides);
+
+      // Reverse the tile along the horizontal, vertical, or both
+      // dimensions
+      auto inputMap = AffineMap::getMultiDimIdentityMap(numDims, context);
+      if (shouldHReflect(horizontalPos)) {
+        inputMap =
+            reflectDim(inputMap, numDims, hDim, getHPadArgument(horizontalPos));
+      }
+      if (shouldVReflect(verticalPos)) {
+        inputMap =
+            reflectDim(inputMap, numDims, vDim, getVPadArgument(verticalPos));
+      }
+
+      tile = rewriter
+                 .create<linalg::GenericOp>(
+                     loc, llvm::cast<RankedTensorType>(tile.getType()), tile,
+                     tile, ArrayRef({inputMap, idMap}), iteratorTypes,
+                     [](OpBuilder &b, Location nestedLoc, ValueRange args) {
+                       b.create<linalg::YieldOp>(nestedLoc, args[0]);
+                     })
+                 .getResult(0);
+
+      // Insert the tile in the resultTensor
```

Comment:
mini-nit: Add period to the end of sentence.

Reply:
Not sure if it is worth to re-run the CI for this though ;-). Anyway, I fixed it.

Reply:
Oh right, my bad. I forgot Torch-MLIR CI takes forever. Thanks for fixing it anyway :D